### PR TITLE
feat: initial support for bun

### PR DESCRIPTION
**Pull request by redbmk**
While `bun create remix` works, I noticed that `bun create remix --template remix-run/indie-stack` was broken. This is likely true of the other stacks as well but I haven't taken a look yet.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +87,6 @@
     "node": ">=18.0.0"
   },
   "prisma": {
-    "seed": "ts-node --require tsconfig-paths/register prisma/seed.ts"
+    "seed": "ts-node -r tsconfig-paths/register prisma/seed.ts"
```
Comment (redbmk) [inline package.json:90]: both `ts-node` and `bun` support `-r`, but of the two only `ts-node` supports `--require` (the long name in `bun` is `--preload`).

File: remix.init/index.js [hunk 2]
```diff
@@ -56,18 +64,45 @@ const getPackageManagerVersion = (packageManager) =>
...
     },
+    devDependencies: { ...devDependencies },
+    prisma: { ...prisma },
   } = packageJson.content;
 
+  if (packageManager.tsNative) {
+    delete devDependencies["ts-node"];
+  }
+
+  const updateScripts = (scripts) => {
```
Comment (redbmk) [inline remix.init/index.js:82]: there are a lot of npm-specific commands in the scripts, so this tries to replace them all with the correct one. This should effectively be a noop if using `npm`.

e.g. `npx ts-node` => `bun`, `npm run` => `yarn`, etc.

**Comment by redbmk** [comment]
There are probably still a ton of optimizations that could be done to make using `bun` a bigger win, but this is quick first pass. This also ends up preferring native package managers over `npm` where possible (not sure if `npm-run-all` would support `pnpm` or `bun`, but it looks like it hasn't been updated in 5 years so I doubt it).

**Comment by MichaelDeBoey** [comment]
Bun support was added in #265

**Pull request closed without merging**